contracts/MoMABurn.sol

```diff
@@ -0,0 +1,179 @@
+    /// @param _burnArtworkID - artwork ID will be burned
+    function setBurnArtworkID(uint256 _burnArtworkID) public onlyOwner {
+        burnArtworkID = _burnArtworkID;
+    }
+
+    /// @notice setMintContracts use for update mint contract addresses
+    /// @param _tier1Address - new burn contract address for tier 1
+    /// @param _tier2Address - new burn contract address for tier 2
+    /// @param _tier3Address - new burn contract address for tier 3
+    function setMintContracts(
+        address _tier1Address,
+        address _tier2Address,
+        address _tier3Address
+    ) public onlyOwner {
+        require(
+            _tier1Address != address(0) &&
+                _tier2Address != address(0) &&
+                _tier3Address != address(0),
+            "Invalid contract address"
+        );
+        tier1ContractAddress = _tier1Address;
+        tier2ContractAddress = _tier2Address;
+        tier3ContractAddress = _tier3Address;
+    }
+
+    /// @notice setMintArtworkIDs use for update mint artwork ID
+    /// @param _tier1ArtworkID - new artwork ID for tier 1
+    /// @param _tier2ArtworkID - new artwork ID for tier 2
+    /// @param _tier3ArtworkID - new artwork ID for tier 3
+    function setMintArtworkIDs(
+        uint256 _tier1ArtworkID,
+        uint256 _tier2ArtworkID,
+        uint256 _tier3ArtworkID
+    ) public onlyOwner {
+        tier1ArtworkID = _tier1ArtworkID;
+        tier2ArtworkID = _tier2ArtworkID;
+        tier3ArtworkID = _tier3ArtworkID;
+    }
+
+    /// @notice setBurnEnabled use for update burn enable status
+    /// @param _isBurnEnabled - the condition is true or false
+    function setBurnEnabled(bool _isBurnEnabled) public onlyOwner {
+        isBurnEnabled = _isBurnEnabled;
+    }
+
+    /// @notice burnAndMint use for burn old editions and mint new edition
+    /// @param _burnedEditions - array of edition IDs to burn
+    function burnAndMint(uint256[] memory _burnedEditions) external {
+        require(isBurnEnabled, "Burn and mint status is not enabled");
+
+        require(
+            _burnedEditions.length == 3 ||
+                _burnedEditions.length == 6 ||
+                _burnedEditions.length == 9,
+            "Invalid number of burning editions"
+        );
+
+        Artwork memory artwork = FFV2(burnContractAddress).artworks(
+            burnArtworkID
+        );
+
+        for (uint256 i = 0; i < _burnedEditions.length; i++) {
+            // Check duplicate with others
+            for (uint256 j = i + 1; j < _burnedEditions.length; j++) {
+                if (_burnedEditions[i] == _burnedEditions[j]) {
+                    revert("Invalid burning editions");
+                }
+            }
+
+            require(
+                artwork.editionSize > 0 &&
+                    _burnedEditions[i] > burnArtworkID &&
+                    _burnedEditions[i] < burnArtworkID + artwork.editionSize,
+                "Edition ID is not support for burning"
+            );
```

Comment:
```suggestion
            require(
                artwork.editionSize > 0 &&
                    _burnedEditions[i] >= burnArtworkID &&
                    _burnedEditions[i] <= burnArtworkID + artwork.editionSize,
                "Edition ID is not support for burning"
            );
```





According to FFV2.
>
     /// @notice The range of editionNumber should be between 0 (AP) ~ artwork.editionSize
     require(
          editionNumber <= artworks[artworkID].editionSize,
          "edition number exceed the edition size of the artwork"
      );

The editionID would be [ artworkID, artworkID + editionSize ].

Reply:
Updated